docs/src/index.md

```diff
@@ -76,160 +97,72 @@ julia> @btime sum(x -> log(x)^7, $x);
-
-For `@threads`, increasing `nchunks` beyond `nthreads()` typically isn't helpful. This is because it will anyways always create O(`nthreads()`) tasks (i.e. a fixed number), grouping up multiple of our chunks if necessary.
-
-```julia
-julia> @btime uneven_workload_threads($x, $work_load; nchunks=8*nthreads(), chunk_type=:batch);
-  2.081 ms (74 allocations: 7.88 KiB)
-
-julia> @btime uneven_workload_threads($x, $work_load; nchunks=8*nthreads(), chunk_type=:scatter);
-  632.149 μs (75 allocations: 7.91 KiB)
-```
-
-### Using `@spawn`
-
-We can use `@spawn` to get "proper" load balancing through Julia's task scheduler. The spawned tasks, each associated with a chunk of the `work_load` array, will be dynamically scheduled at runtime. If there are enough tasks/chunks, the scheduler can map them to Julia threads in such a way that the overall workload per Julia thread is balanced.
-
-Here is the implementation that we'll consider.
-
-```julia
-julia> function uneven_workload_spawn(x, work_load; nchunks::Int, chunk_type::Symbol)
-           ts = map(chunks(work_load, nchunks, chunk_type)) do (idcs, ichunk)
-               @spawn begin
-                   local s = zero(eltype(x))
-                   for i in idcs
-                       s += sum(log(x[j])^7 for j in 1:work_load[i])
-                   end
-                   s
-               end
-           end
-           return sum(fetch.(ts))
-       end
-```
-
-For `nchunks == Thread.nthreads() == 12`, we expect to see similar performance as for the `@threads` variant above, because we're creating the same (number of) chunks/tasks.
-
-```julia
-julia> @btime uneven_workload_spawn($x, $work_load; nchunks=nthreads(), chunk_type=:batch);
-  1.997 ms (93 allocations: 7.30 KiB)
-
-julia> @btime uneven_workload_spawn($x, $work_load; nchunks=nthreads(), chunk_type=:scatter);
-  573.399 μs (91 allocations: 7.23 KiB)
-```
-
-However, by increasing `nchunks > nthreads()` we can give the dynamic scheduler more tasks ("units of work") to balance out and improve the load balancing. In this case, the difference between `:batch` and `:scatter` chunking becomes negligible.
-
-```julia
-julia> @btime uneven_workload_spawn($x, $work_load; nchunks=8*nthreads(), chunk_type=:batch);
-  603.830 μs (597 allocations: 53.30 KiB)
-
-julia> @btime uneven_workload_spawn($x, $work_load; nchunks=8*nthreads(), chunk_type=:scatter);
-  601.519 μs (597 allocations: 53.30 KiB)
-```
+!!! warning
+    Using shared buffers can lead to performance issues caused by *false-sharing*, in which
+    multiple threads try to concurrently read data from nearby memory regions. The example
+    above intends to illustrate the syntax to be used to index the buffers, rather than
+    to suggest this a an optimal pattern for parallelization. 
```

Comment:
```suggestion
!!! warning
    Using shared buffers like this can lead to performance issues caused by *false-sharing*:
    A thread writes to the buffer, invalidates the cache-line for other threads, and thus causes expensive restoration of [cache coherence](https://en.wikipedia.org/wiki/Cache_coherence).
    The example above intends to illustrate the syntax to be used to index the buffers, rather than
    to suggest this a an optimal pattern for parallelization. 
```